tests/libs/util/socket_helper.cpp

```diff
@@ -112,12 +118,37 @@ _base_socket::_base_socket(
     for (int i = 0; i < 5; ++i) {
         error = bind(socket, (PSOCKADDR)&local_addr, sizeof(local_addr));
         if (error == 0) {
+            _bind_succeeded = true;
+            _actual_bind_error = 0;
             break;
         }
-        if (WSAGetLastError() != WSAENOBUFS) {
-            FAIL("Failed to bind socket with error: " << WSAGetLastError());
+
+        _actual_bind_error = WSAGetLastError();
+
+        // If expecting a bind error, capture it and validate later
```

Comment:
```suggestion
        // If expecting a bind error, capture it and validate later.
```
nit: add punctuation to comments here and below.